controllers/complianceeventsapi/server.go

```diff
@@ -0,0 +1,334 @@
+package complianceeventsapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"database/sql"
+	"encoding/hex"
+	"encoding/json"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/doug-martin/goqu/v9"
+	_ "github.com/doug-martin/goqu/v9/dialect/postgres" // blank import the dialect driver
+	_ "github.com/lib/pq"
+)
+
+var (
+	clusterKeyCache      sync.Map
+	parentPolicyKeyCache sync.Map
+	policyKeyCache       sync.Map
+)
+
+type databaseAPIServer struct {
+	Lock      *sync.Mutex
+	server    *http.Server
+	isRunning bool
+}
+
+// Start starts the http server. If it is already running, it has no effect.
+func (s *databaseAPIServer) Start(dbURL string) error {
+	s.Lock.Lock()
+	defer s.Lock.Unlock()
+
+	if s.isRunning {
+		return nil
+	}
+
+	mux := http.NewServeMux()
+
+	s.server = &http.Server{
+		Addr:    ":5480",
+		Handler: mux,
+
+		// need to investigate ideal values for these
+		ReadTimeout:  15 * time.Second,
+		WriteTimeout: 15 * time.Second,
+		IdleTimeout:  15 * time.Second,
+	}
+
+	db, err := sql.Open("postgres", dbURL)
+	if err != nil {
+		return err
+	}
+
+	goquDB := goqu.New("postgres", db)
+
+	// register handlers here
+	mux.HandleFunc("/api/v1/compliance-events", func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodPost:
+			postComplianceEvent(goquDB)(w, r)
+		default:
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		}
+	})
+
+	go func() {
+		err := s.server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			log.Error(err, "Error starting compliance events api server")
+		}
+	}()
+
+	s.isRunning = true
+
+	return nil
+}
+
+// Stop stops the http server. If it is not currently running, it has no effect.
+func (s *databaseAPIServer) Stop() {
+	s.Lock.Lock()
+	defer s.Lock.Unlock()
+
+	if !s.isRunning {
+		return
+	}
+
+	if err := s.server.Shutdown(context.TODO()); err != nil {
+		log.Error(err, "Error stopping compliance events api server")
+	}
+
+	s.isRunning = false
+}
+
+func postComplianceEvent(goquDB *goqu.Database) func(http.ResponseWriter, *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			log.Error(err, "error reading")
+			http.Error(w, "Error reading request body", http.StatusBadRequest)
```

Comment:
I'd prefer if the error return JSON to match successful output and it gives us more flexibility later.

For example, `{"message": "Error reading request body"}`